docarray/doc_index/backends/hnswlib_doc_index.py

```diff
@@ -0,0 +1,400 @@
+
+    @composable
+    def _filter(
+        self,
+        *args,
+        **kwargs,
+    ) -> DocumentArray:
+
+        raise NotImplementedError(
+            f'{type(self)} does not support filter-only queries.'
+            f' To perform post-filtering on a query, use'
+            f' `build_query()` and `execute_query()`.'
+        )
+
+    def _filter_batched(
+        self,
+        filter_queries: Any,
+        limit: int,
+        **kwargs,
+    ) -> List[DocumentArray]:
+        raise NotImplementedError(
+            f'{type(self)} does not support filter-only queries.'
+            f' To perform post-filtering on a query, use'
+            f' `build_query()` and `execute_query()`.'
+        )
+
+    def _text_search(
+        self,
+        query: str,
+        search_field: str,
+        limit: int,
+        **kwargs,
+    ) -> FindResult:
+        raise NotImplementedError(f'{type(self)} does not support text search.')
+
+    def _text_search_batched(
+        self,
+        queries: Sequence[str],
+        search_field: str,
+        limit: int,
+        **kwargs,
+    ) -> FindResultBatched:
+        raise NotImplementedError(f'{type(self)} does not support text search.')
+
+    def __delitem__(self, key: Union[str, Sequence[str]]):
+        # delete from the indices
+        if isinstance(key, str):
+            key = [key]
+        try:
+            for doc_id in key:
+                id_ = self._to_hashed_id(doc_id)
+                for col_name, index in self._hnsw_indices.items():
+                    index.mark_deleted(id_)
+        except RuntimeError:
+            raise KeyError(f'No document with id {key} found')
+
+        self._delete_docs_from_sqlite(key)
+        self._sqlite_conn.commit()
+
+    def __getitem__(self, key: Union[str, Sequence[str]]):
+        # delete from the indices
+        if isinstance(key, str):
+            key = [key]
+        out_docs = self._get_docs_sqlite_doc_id(key)
+        # TODO(johannes): this could be handled for all stores
+        if len(out_docs) == 0:
+            raise KeyError(f'No document with id {key} found')
+        if len(out_docs) == 1:
+            return out_docs[0]
+        return out_docs
+
+    def num_docs(self) -> int:
+        return self._get_num_docs_sqlite()
+
+    ###############################################
+    # Helpers                                     #
+    ###############################################
+
+    # general helpers
+    @staticmethod
+    def _to_hashed_id(doc_id: str) -> int:
+        # https://stackoverflow.com/questions/16008670/how-to-hash-a-string-into-8-digits
+        # hashing to 18 digits avoids overflow of sqlite INTEGER
+        return int(hashlib.sha256(doc_id.encode('utf-8')).hexdigest(), 16) % 10**18
+
+    def _load_index(self, col_name: str, col: '_Column') -> hnswlib.Index:
+        """Load an existing HNSW index from disk."""
+        index = self._create_index_class(col)
```

Comment:
can we create a `hnswlib.Index` with no params instead if it will load content from disk on next line

Reply:
I just read hnswlib documentation. no problem here

Reply:
I'll keep it in mind for the follow-up PR that refines the HNSWlib implementation :)